numalogic/udfs/tools.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from pandas import DataFrame
+
+from numalogic.registry import ArtifactManager, ArtifactData
+from numalogic.tools.exceptions import RedisRegistryError
+from numalogic.tools.types import KEYS
+from numalogic.udfs._config import StreamConf
+from numalogic.udfs.entities import StreamPayload
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_df(
+    data_payload: dict, stream_conf: StreamConf, fill_value: float = 0.0
+) -> tuple[DataFrame, list[int]]:
+    """
+    Function is used to create a dataframe from the data payload.
+    Args:
+        data_payload: data payload
+        stream_conf: stream configuration.
+        fill_value: nan fill value.
+
+    Returns
+    -------
+        dataframe and timestamps
+    """
+    features = stream_conf.metrics
+    df = (
+        pd.DataFrame(data_payload["data"], columns=["timestamp", *features])
+        .fillna(fill_value)
+        .tail(stream_conf.window_size)
+    )
+    return df[features].astype(np.float32), df["timestamp"].astype(int).tolist()
+
+
+def make_stream_payload(
+    data_payload: dict, raw_df: DataFrame, timestamps: list[int], keys: list[str]
+) -> StreamPayload:
+    """
+    Make StreamPayload object
+    Args:
+        data_payload: payload dictionary
+        raw_df: Dataframe
+        timestamps: Timestamps
+        keys: keys.
+
+    Returns
+    -------
+        StreamPayload object
+    """
+    return StreamPayload(
+        uuid=data_payload["uuid"],
+        config_id=data_payload["config_id"],
+        composite_keys=keys,
+        data=np.ascontiguousarray(raw_df, dtype=np.float32),
+        raw_data=np.ascontiguousarray(raw_df, dtype=np.float32),
+        metrics=raw_df.columns.tolist(),
+        timestamps=timestamps,
+        metadata=data_payload["metadata"],
+    )
+
+
+def _load_model(
```

Comment:
isnt this same as load_artifact in inference? why are we not using this in inference?

Reply:
We will move this to the base class in future.

Reply:
but if the logic is same, why repeat in inference?